dqn.hpp

```diff
@@ -13,22 +13,35 @@
 
 namespace dqn {
 
-constexpr auto kStateDataSize = 58;
-constexpr auto kInputCount = 2;
-constexpr auto kInputDataSize = kStateDataSize * kInputCount;
+constexpr auto kInputFrameCount = 2;
 constexpr auto kMinibatchSize = 32;
-constexpr auto kMinibatchDataSize = kInputDataSize * kMinibatchSize;
-constexpr auto kOutputCount = 9;
 
-using StateData = std::array<float, kStateDataSize>;
-using StateDataSp = std::shared_ptr<StateData>;
-using InputStates = std::array<StateDataSp, kInputCount>;
-using Transition = std::tuple<InputStates, int,
-                              float, boost::optional<StateDataSp>>;
+constexpr auto kActorStateDataSize = 58;
+constexpr auto kActorInputDataSize = kActorStateDataSize * kInputFrameCount;
+constexpr auto kActorMinibatchDataSize = kActorInputDataSize * kMinibatchSize;
+constexpr auto kActorOutputCount = 1;
 
-using StateLayerInputData = std::array<float, kMinibatchDataSize>;
-using TargetLayerInputData = std::array<float, kMinibatchSize * kOutputCount>;
-using FilterLayerInputData = std::array<float, kMinibatchSize * kOutputCount>;
+using ActorStateData = std::array<float, kActorStateDataSize>;
+using ActorStateDataSp = std::shared_ptr<ActorStateData>;
+using ActorInputStates = std::array<ActorStateDataSp, kInputFrameCount>;
+using Transition = std::tuple<ActorInputStates, int,
+                              float, boost::optional<ActorStateDataSp>>;
+
+using StateLayerInputData = std::array<float, kActorMinibatchDataSize>;
+using TargetLayerInputData = std::array<float, kMinibatchSize * kActorOutputCount>;
+using FilterLayerInputData = std::array<float, kMinibatchSize * kActorOutputCount>;
+
+constexpr auto kCriticStateDataSize = kActorStateDataSize + 1;
+constexpr auto kCriticInputDataSize = kCriticStateDataSize * kInputFrameCount;
+constexpr auto kCriticMinibatchDataSize = kCriticInputDataSize * kMinibatchSize;
+constexpr auto kCriticOutputCount = 1;
```

Comment:
CriticOutputCount will always equal ActorOutputCount which will always equal the number of continuous actions available.